criu/plugin.c

```diff
@@ -243,7 +244,9 @@ int cr_plugin_init(int stage)
 		if (len < 3 || strncmp(de->d_name + len - 3, ".so", 3))
 			continue;
 
-		snprintf(path, sizeof(path), "%s/%s", opts.libdir, de->d_name);
+		length_needed = snprintf(path, sizeof(path), "%s/%s", opts.libdir, de->d_name);
+		if (length_needed < 0 || (unsigned) length_needed >= sizeof(path))
```

Comment:
length_needed don't include the null byte.
https://linux.die.net/man/3/snprintf

And we need to print an error here.

nit: you can reuse the len variable here.